src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -26,9 +26,17 @@
 #include "lib/core/CHIPError.h"
 
 #define TAG "OTAImageProcessor"
+using namespace chip::System;
 using namespace ::chip::DeviceLayer::Internal;
 
 namespace chip {
+namespace {
+
+void HandleRestart(Layer * systemLayer, void * appState)
+{
+    esp_restart();
```

Comment:
After restart, NotifyUpdateApplied should also be called once the device has booted to a new version. Is that something you intend on doing in a later PR?

Reply:
Yes, I'll add a follow-up PR for that.

Reply:
Added an issue for the same, #13760